pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -712,6 +712,7 @@ public CompletableFuture<Optional<Long>> addProducer(Producer producer,
     }
 
     @Override
+    @Deprecated
```

Comment:
I think we can remove it directly. I don't think any other component will call this method except transactions.